### PR TITLE
API-23726-remove-leading-space-contentions

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Summary: Adds method to handle contentions, and a new test.
- how to reproduce bug: send in an array with leading white spaces
- solution: adds method to split and strip the contentions array if it exists
- team: Dash
- flipper: n/a

## Related issue(s)
- [API-23726](https://vajira.max.gov/browse/API-23726)

## Testing done

- Rspec: added a new test
- Postman:
```
1. {{uri}}/services/claims/v2/veterans/1012826667V821812/claims/600073960
            "contentionList": [
                "coronary artery disease (Increase)"
            ],
2. {{uri}}/services/claims/v2/veterans/1012826667V821812/claims/600021740
            "contentionList": [
                "Coronary Artery Disease (New)"
            ],
3. {{uri}}/services/claims/v2/veterans/1012826667V821812/claims/269744
            "contentionList": null,
4. {{uri}}/services/claims/v2/veterans/1012667145V762142/claims/600360175
            "contentionList": [
                "Meningitis (New)",
                "GERD (New)",
                "bilateral pes planus (New)",
                "gallbladder removal (New)",
                "hearing loss (Increase)",
                "IBS (New)",
                "gunshot wound",
                "left thigh (New)",
                "hepatitis C (New)",
                "Lou Gehrig's Disease (New)",
                "hemorrhoids (New)",
                "asthma (New)",
                "Gastric ulcer (New)",
                "type II diabetes mellitus (New)",
                "endometriosis (New)",
                "leukemia (New)",
                "hypertension (New)",
                "UTI (New)"
            ],
```
## What areas of the site does it impact?
Claims Show

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Comment by FonzMP** [comment]
OAS is showing this in still:
```
"contentionList": [
                        "abnormal heart (New)",
                        " abscess kidney (New)",
                        " encephalitis lethargica residuals (New)",
                        " dracunculiasis (New)",
                        " gingivitis (New)",
                        " abnormal weight loss (New)",
                        " groin condition (New)",
                        " metritis (New)"
                      ]
```

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 1]
```diff
@@ -172,6 +172,14 @@ def build_claim_structure(data:, lighthouse_id:, upstream_id:) # rubocop:disable
           }
         end
 
+        def get_contention_list(data)
+          if data[:contentions].present? && data[:contentions].is_a?(Array)
+            data[:contentions].collect(&:strip)
+          elsif data[:contentions].present? && data[:contentions].is_a?(String)
+            [data[:contentions]]
+          end
+        end
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:198]: is it possible to get an array? my understanding is that we get back a string here that's comma separated. 

I believe we just need to add `.collect(&:strip)` at the end of the old :157.

Comment (stiehlrod) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:198]: I ran across a "contention list" that was a String in the test data, and it caused a 500, so I added a catch for that.

File: modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json [hunk 2]
```diff
@@ -734,14 +734,7 @@
...
                       "contentionList": [
-                        "abnormal heart (New)",
-                        " abscess kidney (New)",
-                        " encephalitis lethargica residuals (New)",
-                        " dracunculiasis (New)",
-                        " gingivitis (New)",
-                        " abnormal weight loss (New)",
-                        " groin condition (New)",
-                        " metritis (New)"
+                        "abnormal heart (New), abscess kidney (New), encephalitis lethargica residuals (New), dracunculiasis (New), gingivitis (New), abnormal weight loss (New), groin condition (New), metritis (New)"
```
Comment (FonzMP) [inline modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json:737]: this is what I'm referring to. the response from BGS is a comma separated value, as a string. In the logic above, you're just adding this string to an array. we want to split and then remove leading (and trailing for that matter) whitespace. Each of these should be their own entry within the array

```
"contentList": [
  "abnormal heart (New)",
  "abscess kidney (New)",
  "enceph.....
]
```

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 3]
```diff
@@ -192,6 +192,15 @@ def build_claim_structure(data:, lighthouse_id:, upstream_id:) # rubocop:disable
           }
         end
 
+        def get_contention_list(data)
+          if data[:contentions].present? && data[:contentions].is_a?(Array)
+            data[:contentions].collect(&:strip)
+          elsif data[:contentions].present? && data[:contentions].is_a?(String)
+            striped_array = data[:contentions].split(',')
```
Comment (mchristiansonVA) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:199]: Very minor/nitpicky: Should this be "stripped_array"?

**Review by mchristiansonVA — APPROVED** [review]
Looks good.

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 4]
```diff
@@ -174,7 +174,7 @@ def build_claim_structure(data:, lighthouse_id:, upstream_id:) # rubocop:disable
             claim_type_code: data[:bnft_claim_type_cd],
             claim_type: data[:claim_status_type],
             close_date: data[:claim_complete_dt].present? ? format_bgs_date(data[:claim_complete_dt]) : nil,
-            contention_list: data[:contentions]&.split(','),
+            contention_list: data[:contentions].present? ? data[:contentions].split(',').map!(&:strip) : nil,
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:177]: let's make this
`data[:contentions]&.split(',')&.collect(&:strip) || []`
VA Gov asked that this not be possible to be nil so we'll want an empty array if nil. And I believe our documentation does not reflect that either. Once those changes are done, we'll need to run the OAS RSWAG updates. This should be an array of items, not an array of one string comma separated. 

Below it shows: 
```
["abnormal heart (New), abscess kidney (New), encephalitis lethargica residuals (New), dracunculiasis (New), gingivitis (New), abnormal weight loss (New), groin condition (New), metritis (New)"]
```
it should be
```
  "abnormal heart (New)",
  "abscess kidney (New)",
  "encephalitis lethargica residuals (New)",
  .......
]
```

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**